src/Symfony/Component/Translation/Translator.php

```diff
@@ -72,32 +72,45 @@ class Translator implements TranslatorInterface, TranslatorBagInterface
     private $configCacheFactory;
 
     /**
+     * @var TranslatorBagInterface
+     */
+    private $translatorBag;
+
+    /**
      * Constructor.
      *
-     * @param string               $locale   The locale
-     * @param MessageSelector|null $selector The message selector for pluralization
-     * @param string|null          $cacheDir The directory to use for the cache
-     * @param bool                 $debug    Use cache in debug mode ?
+     * @param string                 $locale        The locale
+     * @param MessageSelector|null   $selector      The message selector for pluralization
+     * @param string|null            $cacheDir      The directory to use for the cache
+     *                                              Deprecated since version 2.8, to be removed in 3.0.
+     * @param bool                   $debug         Use cache in debug mode ?
+     *                                              Deprecated since version 2.8, to be removed in 3.0.
+     * @param TranslatorBagInterface $translatorBag
      *
      * @throws \InvalidArgumentException If a locale contains invalid characters
      *
      * @api
      */
-    public function __construct($locale, MessageSelector $selector = null, $cacheDir = null, $debug = false)
+    public function __construct($locale, MessageSelector $selector = null, $cacheDir = null, $debug = false, TranslatorBagInterface $translatorBag = null)
```

Comment:
the new signature should be supported in place instead of adding new arguments (look at how this is handled in https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/SecurityBundle/Templating/Helper/LogoutUrlHelper.php for instance